tests/utils.py

```diff
@@ -19,42 +19,25 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import asyncio
-from abc import ABC, abstractmethod
+import os
 
 
-class ButlerIngester(ABC):
-    """Interface class for processing files for ingestion into a butler.
+def removeEntries(directory):
+    """Remove given directory and all entries below it
+    Parameters
+    ----------
+    dictectory: `str`
+        directory to remove
     """
-
-    @abstractmethod
-    def ingest(self, filename):
-        """Placeholder to ingest a file
-
-        Parameters
-        ----------
-        filename: `str`
-            file name to ingest
-        """
-        raise NotImplementedError()
-
-    @abstractmethod
-    def getName(self):
-        """Get the name of this ingester
-
-        Returns
-        -------
-        ret: `str`
-            the name of this ingester
-        """
-        raise NotImplementedError()
-
-    async def run_task(self):
-        """Run task that require periodical attention
-        """
-        await asyncio.sleep(60)
-
-    def clean(self):
-        """Perform a cleaning pass for this ingester; override if necessary
-        """
-        pass
+    if os.path.exists(directory) is False:
+        return
+    if os.path.isdir(directory) is False:
+        os.unlink(directory)
+        return
+    with os.scandir(directory) as entries:
+        for ent in entries:
+            if os.path.isdir(ent):
+                removeEntries(ent)
+            else:
+                os.unlink(ent)
+    os.rmdir(directory)
```

Comment:
Is there a reason behind removing the directory tree "manually" (lines 37-43) instead of using `shutil.rmtree()`?

Reply:
that's a better way to do it.  I had been looking at os.rmdir, which doesn't work if the directory is empty.